File: crates/nil/src/commands/round.ts
import { invoke } from '@tauri-apps/api/core';

export function getRound() {
  return invoke<Round>('get_round');
}

export function isRoundIdle() {
  return invoke<boolean>('is_round_idle');
}

export function startRound() {
  return invoke<null>('start_round');
}

File: crates/nil/src-tauri/src/command/round.rs
use crate::error::Result;
use crate::manager::ManagerExt;
use nil_core::round::Round;
use tauri::AppHandle;

#[tauri::command]
pub async fn get_round(app: AppHandle) -> Result<Round> {
  app
    .client(async |cl| cl.round().await)
    .await?
    .map_err(Into::into)
}

#[tauri::command]
pub async fn is_round_idle(app: AppHandle) -> Result<bool> {
  app
    .client(async |cl| cl.round().await)
    .await?
    .map(|round| round.is_idle())
    .map_err(Into::into)
}

#[tauri::command]
pub async fn start_round(app: AppHandle) -> Result<()> {
  app
    .client(async |cl| cl.start_round().await)
    .await?
    .map_err(Into::into)
}
